latin_word_syllabification.py

```diff
@@ -0,0 +1,337 @@
+    vowel_positions = [idx for idx, char in enumerate(word) if char in _VOWELS]
+    return vowel_positions
+
+
+def _get_syl_bound_position(ltrs_btw_vow_grps: str) -> "tuple[int, str]":
+    """
+    Find the adjustment required to a syllable boundary between
+    two vowel groups based on the letters between them.
+
+    We handle 4 general cases.
+
+    1. No consonants between vowel groups: keep the syllable boundary
+          where it is diaeresis unless we have vowel + i + vowel,
+          in which case we split as [vowel] + [i + vowel]
+    2. 1 consonant between vowel groups: keep the syllable boundary
+          where it is (consonant is part of second syllable)
+    3. 2 consonants between vowel groups: split the first consonant to  the
+          first syllable, unless the two consonants form a consonant group, in
+          which case keep the group on the second syllable
+    4. 3+ consonants between vowel groups: add the first consonant or
+          consonant group to the first syllable
+
+    Two additional special cases exist. "X" is treated as a double consonant
+    "ks" and the letter terminates the previous syllable. In cases where "i"
+    is between two vowels, but not part of a dipthong, it is treated as a
+    consonant and begins the following syllable.
+
+    ltrs_btw_vow_grps [str]: letters between two vowel groups
+
+    returns [tuple[int, str]]: tuple of the form (syl_bound, split_case),
+        where syl_bound is the index of the syllable boundary relative to the
+        start of ltrs_btw_vow_grps (eg. 0 = syllable boundary before
+        first letter, 1 = syllable boundary between first and second letters, etc.)
+        and split_case is a string describing the case used to determine the
+        syllable boundary (passed to logger).
+    """
+    num_ltrs_btw_vow_grps = len(ltrs_btw_vow_grps)
+    # Default case: syllable boundary immediately follows previous
+    # vowel group.
+    syl_bound = 0
+    if num_ltrs_btw_vow_grps == 0:
+        split_case = "Hiatus"
+    elif ltrs_btw_vow_grps[0] == "x":
+        syl_bound = 1
+        split_case = "X is double consonant"
+    elif num_ltrs_btw_vow_grps == 1:
+        split_case = "1 consonant between vowels"
+    elif num_ltrs_btw_vow_grps == 2:
+        if ltrs_btw_vow_grps not in _CONSONANT_GROUPS:
+            syl_bound = 1
+            split_case = "2 consonants between vowels"
+        else:
+            split_case = "2 consonants between vowels (consonant group)"
+    else:
+        if ltrs_btw_vow_grps[:2] in _CONSONANT_GROUPS:
+            syl_bound = 2
+        else:
+            syl_bound = 1
+        split_case = "3+ consonants between vowels"
+    return syl_bound, split_case
+
+
+def _syllabify(word: str) -> "list[int]":
+    """
+    Finds indices of the syllable boundaries of a word.
+    See README for details on syllabification rules.
+
+    word [str]: word to syllabify
+
+    returns [list[int]]: list of syllables
```

Comment:
`returns [list[int]]: list of string indices indicating where syllables in word begin`, no?